napari/utils/_tests/test_migrations.py

```diff
@@ -83,3 +85,49 @@ class macOS:
 
         class MacOSXServer(MacOSX):
             pass
+
+
+def test_deprecating_dict_get_deprecated_key():
+    d = DeprecatingDict({'a': 1, 'b': 2})
+    d.deprecate(
+        DeprecatedParameter(
+            previous_name='c',
+            new_name='a',
+            version='v2.0',
+            since_version='v1.6',
+        )
+    )
+    with pytest.warns(FutureWarning):
+        value = d['c']
+    assert value == d['a']
+
+
+def test_deprecating_dict_set_deprecated_key():
+    d = DeprecatingDict({'a': 1, 'b': 2})
+    d.deprecate(
+        DeprecatedParameter(
+            previous_name='c',
+            new_name='a',
+            version='v2.0',
+            since_version='v1.6',
+        )
+    )
+    with pytest.warns(FutureWarning):
+        d['c'] = 3
+        assert d['c'] == 3
```

Comment:
We should also test value of `d['a']` here, also `with pytest.warns(FutureWarning):` should handle only single line

Reply:
@Czaki do you mean this?

```suggestion
    with pytest.warns(FutureWarning):
        d['c'] = 3
        assert d['c'] == 3
        assert d['c'] == d['a']
```

Reply:
(and if so, feel free to commit the suggestion)

Reply:
```suggestion
    with pytest.warns(FutureWarning):
        d['c'] = 3
   with pytest.warns(FutureWarning):
        assert d['c'] == 3
   assert d['a'] == 3
```
But the last line will not work with the current implementation. 

Having two lines under `with pytest.warns(FutureWarning):` pass if any of the lines emit warning.

Reply:
Will update the tests to have separate `with pytest.warns`.

But I disagree about asserting `d['a'] == 3` because not all deprecations are renames / direct replacements.

So I don't think we should always try to enforce consistency between the value associated with the deprecated key and related non-deprecated one.

If the user sets a deprecated key, they will get a warning and we'll do our best not to break them. That seems like enough to me.

Reply:
Another way of explaining my line of thinking here is that the current implementation of `DeprecatingDict` mostly just behaves like a `dict`. The exception is when you use a deprecated key, in which case it will treat those keys separately and will issue warnings. It will not change the behavior of `dict` when using non-deprecated keys under any circumstances.

By trying to enforce consistency between deprecated and non-deprecated keys, interacting with the deprecated keys has the potential side effect of mutating non-deprecated items. While this might be beneficial in some cases, I think thoses cases are limited in real world usage and I think the extra complication of deviating from `dict` is not worth it - especially since some deprecated keys may not have direct replacements.